### ISSUE TITLE
Compiling the plugin failed

### ISSUE THREAD

**Issue by tonyho** [wangzw/CppStyle#47]
Right now the cdt(`http://download.eclipse.org/tools/cdt/releases/9.3`) repo required at pom.xml is missing.
Using the `mvn clean install` would fail.

**Comment by RustinHaaseCSS** [comment]
I can confirm this, has anyone gotten it to work?

**Comment by sylt** [comment]
From what I've seen, it comes from that the archived CDT have broken links inside their xml-files referring to the various CDT patch releases; only the most recent CDT patch release seems to be available in the archive. So in the case of CppStyle, it refers to CDT 9.3, which has 3 patch releases, while only 1 is available (9.3.3). I have [reported an issue](https://github.com/eclipse-cdt/cdt/issues/235) for this, but no feedback so far.

So, in the meantime, a workaround could be to refer to a release which doesn't have multiple patch releases, such as 9.6. With that, I'm able to progress the build, to the point where I can start compiling. I personally get some strange errors, but I'm suspecting that it's just me and my set up. 

By modifying pom.xml to use 9.6 instead of 9.3 as release, are you able to get the plugin compiling correctly?

**Comment by RustinHaaseCSS** [comment]
Thanks, I was able to compile but the plugin causes the C/C++ option in preferences to disappear.(Using the STM32 version of eclipse). 
I used Java 19.0.1 to compile because that is what I had installed.

This seems to be the same behavior as from one of the download copies, maybe a patch is needed to support this version of eclipse? This strikes me as getting out of scope on this issue though.

My Eclipse Version
```
STM32CubeIDE

Version: 1.11.0
Build: 13638_20221122_1308 (UTC)
```

**Comment by sylt** [comment]
> Thanks, I was able to compile but the plugin causes the C/C++ option in preferences to disappear.(Using the STM32 version of eclipse).

Interesting, I have a colleague which experiences the very same problem as you describe. For me however, when running STM32CubeIDE 0.11.0, I just get an error on my first restart, and after that, everything works as normal with the plugin installed and working as intended. I haven't been able to figure out why it would work for me and not for him... My colleague shared a very long stack trace with me that he got from STM32CubeIDE, but unfortunately, none of the stack frames contained anything referring to the plug-in directly. 

> This seems to be the same behavior as from one of the download copies, maybe a patch is needed to support this version of eclipse?

Alright, that's very good to know! Looking at the version history of this repository, it seems as if the bulk work happened between 2014-2017, and things have probably changed since then. Perhaps if the original developer is busy with other stuff for the time being, it would be an idea to fork the repository/development until they come back.

> This strikes me as getting out of scope on this issue though.

Yes, maybe you are right. At least the patch in [this comment](https://github.com/wangzw/CppStyle/issues/47#issuecomment-1381812679) makes things build for people wanting to contribute.

**Comment by sylt** [comment]
@tonyho: Now that the eclipse-cdt issue is fixed, does this work now for you?

**Comment by sylt** [comment]
@RustinHaaseCSS: Regarding STM32CubeIDE, it seems as if ST released a new version, 1.11.2, which apparently fixes some misc plugin installation issue (see [CHANGELOG](https://www.st.com/resource/en/release_note/rn0114-stm32cubeide-release-v1112-stmicroelectronics.pdf)). It might be interesting to check it out!

If that's the case, I guess that all open questions in this issue should be resolved, since it should now be possible to build and install (although manually for the time being).

**Comment by wangzw** [comment]
May be fixed, CI passed. Reopen if not.
